backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -1054,19 +1046,36 @@ abstract class ObjCExportHeaderGenerator internal constructor(
         return ObjCExportedInterface(generatedClasses, extensions, topLevel, headerLines, namer, mapper)
     }
 
+    fun getExportStubs(): ObjCExportedStubs =
+        ObjCExportedStubs(classForwardDeclarations, protocolForwardDeclarations, stubs)
+
     protected abstract fun reportWarning(text: String)
 
     protected abstract fun reportWarning(method: FunctionDescriptor, text: String)
 
     protected open fun getAdditionalImports(): List<String> = emptyList()
 
-
-    fun translateModule(): List<Stub<*>> {
+    fun translateModule() {
         // TODO: make the translation order stable
         // to stabilize name mangling.
+        translateBaseDeclarations()
```

Comment:
If your intention is to translate modules separately, then you would end up with same "base declaration" in all translation results. Can this be a problem?

Reply:
No, we don't call this method (except when we do want to write a full header to disk). We currently call the more fine-granular methods directly.
But thinking about it, I guess we don't need this much granularity, so I propose we reduce visibility for `translatePackageFragments` and `translateExtraClasses` and introduce a method `translateModuleDeclarations`, which calls both of them (but doesn't unnecessarily couple us to these fine-granular methods).